src/sentry/utils/snuba.py

```diff
@@ -1144,16 +1142,17 @@ def get_json_type(snuba_type):
         return "string"
 
     # Ignore Nullable part
-    nullable_match = re.search(r"^Nullable\((.+)\)$", snuba_type)
+    if snuba_type.startswith("Nullable("):
+        snuba_type = snuba_type[9:-1]
 
-    if nullable_match:
-        snuba_type = nullable_match.group(1)
-
-    # Check for array
-    array_match = re.search(r"^Array\(.+\)$", snuba_type)
-    if array_match:
+    if snuba_type.startswith("Array("):
         return "array"
 
+    # timestamp is DateTime, whereas toStartOf{Hour,Day} are
+    # DateTime('UTC') or DateTime('Universal')
+    if snuba_type.startswith("DateTime("):
```

Comment:
@alxu-sentry Can you add a test for this here? https://github.com/getsentry/sentry/blob/658504a9b2cab9b2c2578cd2e0f474817819a7f9/tests/sentry/utils/test_snuba.py#L147-L159